internal/controller/storagemgr/storagedriver/storagehandler.go

```diff
@@ -76,6 +84,46 @@ func (handler StorageHandler) addContext(next func(http.ResponseWriter, *http.Re
 	})
 }
 
+// processGetAsyncRequest is used to handle Async Get Requests
+func (handler StorageHandler) processGetAsyncRequest(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	deviceName := vars[deviceNameKey]
+	resourceName := vars[resourceNameKey]
+
+	handler.logger.Debug(fmt.Sprintf("Received POST for Device=%s Resource=%s", deviceName, resourceName))
+
+	_, err := handler.service.GetDeviceByName(deviceName)
+	if err != nil {
+		handler.logger.Error(fmt.Sprintf("Incoming reading ignored. Device '%s' not found", deviceName))
+		http.Error(writer, fmt.Sprintf("Device not found"), http.StatusNotFound)
+		return
+	}
+	_, ok := handler.service.DeviceResource(deviceName, resourceName, "get")
+	if !ok {
+		handler.logger.Error(fmt.Sprintf("Incoming reading ignored. Resource '%s' not found", resourceName))
+		http.Error(writer, fmt.Sprintf("Resource not found"), http.StatusNotFound)
+		return
+	}
+
+	ServerIP, err := getServerIP(configPath)
+
+	if err != nil {
+		http.Error(writer, fmt.Sprintf("Configuration File Not Found"), http.StatusNotFound)
+		return
+	}
+
+	RequestUrl := "http://"+ServerIP+":48080/api/v1/reading/name/"+resourceName+"/device/"+deviceName+"/1"
+	log.Println(RequestUrl)
+	resp, err := http.Get(RequestUrl)
+	if err != nil {
+		http.Error(writer, fmt.Sprintf("Resource not found"), http.StatusNotFound)
+		return
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	fmt.Fprintf(writer,string(bodyBytes))
+}
+
 func (handler StorageHandler) processAsyncRequest(writer http.ResponseWriter, request *http.Request) {
```

Comment:
I think it would be good if this function name is changed to `processAsyncPostRequest`.
What do you think?

Reply:
Sure, good Idea !